src/main/java/com/suse/saltstack/netapi/event/BeaconEvent.java

```diff
@@ -0,0 +1,46 @@
+package com.suse.saltstack.netapi.event;
+
+import com.suse.saltstack.netapi.datatypes.Event;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class BeaconEvent {
+    private static final Pattern ret = Pattern.compile("^salt/beacon/([^/]+)/([^/]+)/(.+)$");
+
+    private final String beacon;
+    private final String minionId;
+    private final String path;
+    private final Map<String, Object> data;
+
+    public BeaconEvent(String minionId, String beacon, String path, Map<String, Object> data) {
+        this.minionId = minionId;
+        this.beacon = beacon;
+        this.path = path;
+        this.data = data;
+    }
+
+    public String getBeacon() {
+        return beacon;
+    }
+
+    public String getMinionId() {
+        return minionId;
+    }
+
+    public Map<String, Object> getData() {
+        return data;
+    }
+
+    public static Optional<BeaconEvent> parse(Event event) {
+        Matcher m = ret.matcher(event.getTag());
```

Comment:
@AdamMajer I would say, this example is a bit out of the context, because you use integer temporary variables, that are actually allowed for historical reasons and used as an [exceptions from earlier times](https://www.cwu.edu/~gellenbe/javastyle/variable.html).

Still, as per [Oracle Java Code Conventions](http://www.oracle.com/technetwork/java/codeconventions-135099.html), _"one-character variable names should be **avoided** except for temporary "throwaway" variables. Common names for temporary variables are `i`, `j`, `k`, `m`, and `n` for **integers**; `c`, `d`, and `e` for **characters**"_. The Matcher instance is nor an integer neither character.

Also Google [avoiding](https://google.github.io/styleguide/javaguide.html#s5.2.7-local-variable-names) all one-character local variables.

Also in the Salt itself rules are at least three characters for _everything_, otherwise PyLint will just refuse your code.

So I think we should stick to that too. So @lucidd, please fix that. :wink: